# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/jit/TrialInlining.h
# Commit: dab916754456
# Full Hash: dab916754456373246c85d658fde0520c7246865
# Author: Iain Ireland <iireland@mozilla.com>
# Date: 2020-07-23 03:20:25
# Regressor Bug: 1646378
# File Overlap Count: 1
# Description:
#   Bug 1646378: Invalidate the root script in MaybeInvalidateWarp r=jandem
#   
#   When we replace a stub in an inlined ICScript, we want to invalidate the script into which it was inlined, rather than the original copy of the inlined script.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D84431
# ==============================================================================

diff -r 86db033f2b35 -r dab916754456 js/src/jit/TrialInlining.h
--- a/js/src/jit/TrialInlining.h	Wed Jul 22 11:31:25 2020 +0000
+++ b/js/src/jit/TrialInlining.h	Wed Jul 22 21:57:45 2020 +0000
@@ -41,7 +41,8 @@
  */
 class InliningRoot {
  public:
-  explicit InliningRoot(JSContext* cx) : inlinedScripts_(cx) {}
+  explicit InliningRoot(JSContext* cx, JSScript* owningScript)
+      : owningScript_(owningScript), inlinedScripts_(cx) {}
 
   FallbackICStubSpace* fallbackStubSpace() { return &fallbackStubSpace_; }
 
@@ -53,8 +54,11 @@
 
   void purgeOptimizedStubs(Zone* zone);
 
+  JSScript* owningScript() const { return owningScript_; }
+
  private:
   FallbackICStubSpace fallbackStubSpace_ = {};
+  HeapPtr<JSScript*> owningScript_;
   js::Vector<js::UniquePtr<ICScript>> inlinedScripts_;
 };
 